boards/scdz/ch32v307evt/doc/index.rst

```diff
@@ -0,0 +1,91 @@
+.. zephyr:board:: ch32v307evt
+
+Overview
+********
+
+The `SCDZ`_ CH32V307EVT hardware provides support for QingKe 32-bit RISC-V4F
```

Comment:
This looks like it was adapted from the other WCH READMEs. These were recently updated to fix some of the wording and style - could you pull in the same changes please?